pygromos/files/blocks/imd_blocks.py

```diff
@@ -791,18 +791,21 @@ def read_content_from_str(self, content: List[str]):
             setattr(self, self._order[0][2][0], int(content[3].split()[0]))
             TEMP0 = []
             TAU = []
+            # Read in the temperatures of each temp bath
             for ind in range(5, 5+self.NBATHS):
                 fields = list(map(float, content[ind].split()))
                 TEMP0.append(fields[0])
                 TAU.append(fields[1])
             setattr(self, "TEMP0", TEMP0)
             setattr(self, "TAU", TAU)
+            # Read in DOFSET
             setattr(self, self._order[0][4][0], int(content[5+self.NBATHS+1].split()[0]))
-
+            
             LAST = []
             COMBATH=[]
             IRBATH=[]
-            for ind in range(10, 5+3+2*self.NBATHS):
+            # Read in the last atoms of each temp bath
+            for ind in range(5+3+self.NBATHS, 5+3+2*self.NBATHS):
```

Comment:
This is difficult to follow for me, but I think you checked the general correctness here.